dialect/mysql/starters.go

```diff
@@ -100,3 +103,30 @@ func Raw(query string, args ...any) Expression {
 func Cast(exp bob.Expression, typname string) Expression {
 	return bmod.Cast(exp, typname)
 }
+
+type CaseChain[T bob.Expression] func() expr.Case[T]
```

Comment:
If possible, try to move `CaseChain` to `expr`.

This may involve some type parameter trickery to get it to return `mysql.Expression` from `Else()` and `End()`

However, this will prevent duplication in `psql` and `sqlite`